From d2c21466b94ecb7456975a62c9ae2fb0732daf98 Mon Sep 17 00:00:00 2001
From: Piotr Nowak <pn@semihalf.com>
Date: Thu, 27 Sep 2012 10:00:39 +0200
Subject: [PATCH 314/609] PM, sdio: Add support to SDIO resume on ArmadaXP-B0.

Call SDIO windows reinitialization on board wake-up routine.
No additional actions were required. SDIO interface was operative
after resume from suspend to mem mode unless it was previously
unmounted.

Signed-off-by: Piotr Nowak <pn@semihalf.com>
Signed-off-by: Seif Mazareeb <seif@marvell.com>
---
 arch/arm/mach-armadaxp/core.c |    9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm/mach-armadaxp/core.c b/arch/arm/mach-armadaxp/core.c
index e4e52f0..de828cf 100755
--- a/arch/arm/mach-armadaxp/core.c
+++ b/arch/arm/mach-armadaxp/core.c
@@ -1475,6 +1475,15 @@ void axp_db_restore(void)
 		return;
 	}
 
+#ifdef CONFIG_MV_INCLUDE_SDIO
+	static MV_UNIT_WIN_INFO addrWinMap[MAX_TARGETS + 1];
+
+	if(MV_OK == mvCtrlAddrWinMapBuild(addrWinMap, MAX_TARGETS + 1))
+		if (MV_OK == mvSdmmcWinInit(addrWinMap)) {
+			printk("Resuming SDIO\n");
+		}
+#endif
+
 #ifdef CONFIG_MV_INCLUDE_USB
 	mvUsbPllInit();
 #endif
-- 
1.7.9.5

